app/Libraries/Wiki/SitemapRenderer.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ *    Copyright (c) ppy Pty Ltd <contact@ppy.sh>.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace App\Libraries\Wiki;
+
+use App\Libraries\Elasticsearch\Search;
+use App\Libraries\Elasticsearch\Sort;
+use App\Libraries\Search\BasicSearch;
+
+class SitemapRenderer extends MarkdownRenderer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($page, $body)
+    {
+        parent::__construct($page, $body);
+
+        $this->fetchPages();
+        $this->generateSitemap();
+    }
+
+    private function fetchPages()
+    {
+        $pages = [];
+        $titles = [];
+
+        $cursor = [''];
+
+        while ($cursor !== null) {
+            $search = $this->newBaseSearch()->searchAfter(array_values($cursor));
+            $response = $search->response();
+
+            foreach ($response as $hit) {
+                $path = $hit->source('path');
+                $title = $hit->source('title');
+
+                $pages[] = [
+                    'title' => $hit->source('title'),
+                    'path' => $path,
+                    'depth' => substr_count($path, '/'),
+                ];
+
+                $titles[$path] = $title;
+            }
+
+            $cursor = $search->getSortCursor();
+        }
+
+        // sorts the pages first by the respective parts of their path,
+        // and then, if necessary by their depth (so a parent page
+        // will end up before its children)
+        usort($pages, function ($left, $right) use ($titles) {
```

Comment:
why not just sort by path? `$pages = collect($pages)->sortBy('path')->toArray();`